stock_picking_backorder_no_responsible/models/stock_picking.py

```diff
@@ -0,0 +1,16 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, api
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.model_create_multi
+    def create(self, vals):
+        res = super(StockPicking, self).create(vals)
+        context = self.env.context
+        if context.get('skip_backorder') == True and context.get('cancel_backorder') == False and res.user_id:
+            res.user_id = False
+        return res
```

Comment:
```suggestion
    @api.model_create_multi
    def create(self, vals_list):
        context = self.env.context
        if context.get("skip_backorder") and not context.get("cancel_backorder"):
            for vals in vals_list:
                vals["user_id"] = False

        return super().create(vals_list)
```
Instead of creating the records and then updating `user_id` afterwards, it might be simpler to handle it this way.

Reply:
The proposed `if` statement is incorrect. The `get` function returns `true` if it finds a key with the name `X` in the context. If it finds it, it will return `true`; otherwise, it will return `empty`. In other words, the return value of `get` (whether or not it finds the field) is one thing, and its value is another.

On the other hand, I think it's correct to check in the `if` statement whether the `user_id` field is filled, and if not, do nothing. Why would we clear an already empty field?

Reply:
Regarding to `context.get()` controversy, both approaches are equivalent IMO. In fact, Dante's approach is more generic (e.g. if `skip_backorder` key value is `True`, but also `"pepe"`, it will pass `if` statement).

In the other hand, `user_id` value set... we'd prefer filling every single `vals` with proper value before calling `create`, as OCA recommends it, rather than updating already saved values.

And, at last, replace `vals` by `vals_list` at method definition, if possible.